### ISSUE TITLE
IndexError in var smoothed in get_posteriors 

### ISSUE THREAD

**Issue by mkrech** [k-sys/covid-19#22]
No data in var smoothed with example csv attached.

Hotfix :) :
```
def prepare_cases(cases):
    new_cases = cases.diff()

    smoothed = new_cases.rolling(9,
        win_type='gaussian',
        min_periods=1,
        center=True).mean(std=3).round()
    
    smoothed_tmp = smoothed
    for i in range(10, 0, -1):
        idx_start = np.searchsorted(smoothed, i)
        smoothed_tmp = smoothed.iloc[idx_start:]
        original = new_cases.loc[smoothed_tmp.index]
        if len(smoothed_tmp) > 0:
            break 
        
    return original, smoothed_tmp
```

#########
[IndexError  ] [Traceback](url
[test2.csv.zip](https://github.com/k-sys/covid-19/files/4500003/test2.csv.zip)

) (most recent call last)
~/Projekte/_work/covid-19/Realtime-R0_jhu.py in 
     501 # Note that we're fixing sigma to a value just for the example
     502 
---> 503 posteriors, log_likelihood = get_posteriors(smoothed, sigma=.25)

~/Projekte/_work/covid-19/Realtime-R0_jhu.py in get_posteriors(sr, sigma)
     472         index=r_t_range,
     473         columns=sr.index,
---> 474         data={sr.index[0]: prior0}
     475     )
     476 

~/opt/miniconda3/envs/corona/lib/python3.7/site-packages/pandas/core/indexes/extension.py in __getitem__(self, key)
    207 
    208     def __getitem__(self, key):
--> 209         result = self._data[key]
    210         if isinstance(result, type(self._data)):
    211             return type(self)(result, name=self.name)

~/opt/miniconda3/envs/corona/lib/python3.7/site-packages/pandas/core/arrays/datetimelike.py in __getitem__(self, key)
    512         getitem = self._data.__getitem__
    513         if is_int:
--> 514             val = getitem(key)
    515             if lib.is_scalar(val):
    516                 # i.e. self.ndim == 1

IndexError: index 0 is out of bounds for axis 0 with size 0

**Comment by Bhargavamacha** [comment]
It's happening because smoothed is empty. I faced the same problem. I traced the issue to a few states having lesser cases than the algorithm can work with. Removing those states will solve the problem.

**Comment by k-sys** [comment]
Yes - the cause is empty series when the region you’re analyzing doesn’t meet a threshold. You could remove threshold but I wouldn’t because you’re probably analyzing too little data

Kevin
